tests/e2e-pw/tests/permalink-settings/permalink-settings.block_theme.spec.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { test, expect } from '@woocommerce/e2e-playwright-utils';
+import { cli } from '@woocommerce/e2e-utils';
+
+/**
+ * Internal dependencies
+ */
+import { goToShop, addToCart } from '../../utils';
+
+test.describe(
+	'Tests permalink settings for the cart and checkout templates',
+	async () => {
+		test.afterAll( async () => {
+			await cli(
+				'npm run wp-env run tests-cli "wp option update woocommerce_cart_page_endpoint cart"'
+			);
+			await cli(
+				'npm run wp-env run tests-cli "wp option update woocommerce_checkout_page_endpoint checkout"'
+			);
+		} );
+
+		test.describe( 'Settings page', () => {
+			test( 'Load advanced settings', async ( { page } ) => {
+				await page.goto(
+					'/wp-admin/admin.php?page=wc-settings&tab=advanced'
+				);
+				const title = page
+					.locator( 'div.wrap.woocommerce > form > h2' )
+					.first();
+				await expect( title ).toHaveText( 'Page setup' );
+			} );
+			test( 'Permlink settings exist', async ( { page } ) => {
+				await page.goto(
+					'/wp-admin/admin.php?page=wc-settings&tab=advanced'
+				);
+				const cartInput = page.locator(
+					'tr:has-text("Cart page "):has-text("cart template") input'
+				);
+				const checkoutInput = page.locator(
+					'tr:has-text("Checkout page "):has-text("checkout template") input'
+				);
+
+				await expect( cartInput ).toBeVisible();
+				await expect( checkoutInput ).toBeVisible();
+			} );
+		} );
+
+		test.describe( 'Frontend templates are updated', () => {
+			test.beforeEach( async ( { page } ) => {
+				await goToShop( page );
+				await addToCart( page );
+			} );
+
+			test( 'Changing cart permalink works', async ( { page } ) => {
+				await page.goto(
+					'/wp-admin/admin.php?page=wc-settings&tab=advanced'
+				);
+				const cartInput = page.locator(
+					'tr:has-text("Cart page "):has-text("cart template") input'
```

Comment:
Just noting that I would love to revisit this and use the [provided locators](https://playwright.dev/docs/api/class-locator#locator-get-by-label) in future. For now this is OK to get tests working.